pkg/resource/deploy/source_eval.go

```diff
@@ -2068,6 +2068,28 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 			providers.SetProviderURL(props, opts.GetPluginDownloadUrl())
 		}
 
+		if req.GetPackageRef() != "" {
```

Comment:
Maybe just a quick comment here to explain the block like we have for the other blocks?